vermouth/processors/do_mapping.py

```diff
@@ -229,23 +229,27 @@ def modification_matches(molecule, mappings):
         A list with the following items:
             Dict describing the correspondence of node keys in `molecule` to
                 node keys in the modification.
+
             The modification.
-                Dict with all reference atoms, mapping modification nodes to
-                    nodes in `molecule`.
+
+            Dict with all reference atoms, mapping modification nodes to
+                nodes in `molecule`.
     """
     modified_nodes = set()  # This will contain whole residues.
     for idx, node in molecule.nodes.items():
         if node.get('modifications', []):
             modified_nodes.add(idx)
     ptm_subgraph = molecule.subgraph(modified_nodes)
     grouped = nx.connected_components(ptm_subgraph)
-    found_ptm_groups = set()
+    found_ptm_groups = []
```

Comment:
Why not a set anymore?

Reply:
It's now turned into a list of sets, rather than a set of tuples.
The inner set is now a set of names of all modifications that are in the group of PTMs. So you end up with a list of all sets of names, which together describe all groups.